testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication.py

```diff
@@ -3896,6 +3897,115 @@ def test_blob_contructor_replication(params_from_base_test_setup, blob_data_type
+
+    # 3a. create a cbl db, but do not create any doc (cbl db is empty)
+    cbl_db_name = "empty-db-" + str(time.time())
+    cbl_db = db.create(cbl_db_name, db_config)
+    # 3b. alternate scenario - create docs on cbl then delete all docs on cbl
+    if cbl_action_before_init_replication:
+        db.create_bulk_docs(db=cbl_db, number=9, id_prefix="pre-repl", channels=channels)
+        cbl_doc_ids = db.getDocIds(cbl_db)
+        db.delete_bulk_docs(cbl_db, cbl_doc_ids)
+
+    # 4. create a pull replicator, start replication
+    replicator = Replication(base_url)
+    authenticator = Authenticator(base_url)
+    replicator_authenticator = authenticator.authentication(session_id, cookie, authentication_type="session")
+    repl = replicator.configure_and_replicate(source_db=cbl_db,
+                                              target_url=sg_blip_url,
+                                              continuous=False,
+                                              replicator_authenticator=replicator_authenticator,
+                                              replication_type="pull")
+
+    # 5. verify SGW stats, cbl_replication_pull.rev_send_count sent to CBL equals to 90
+    expvars = sg_client.get_expvars(sg_admin_url)
+    rev_send_count = expvars["syncgateway"]["per_db"][sg_db]["cbl_replication_pull"]["rev_send_count"]
+    if cbl_action_before_init_replication:
+        assert rev_send_count == 100, "rev_send_count {} is incorrect".format(rev_send_count)
+    else:
+        assert rev_send_count == 90, "rev_send_count {} is incorrect".format(rev_send_count)
+
+    # 6. verify CBL, total number of docs is 90, and verify the doc ids are not belongs to tombstone docs
+    cbl_doc_ids = db.getDocIds(cbl_db)
+    cbl_docs = db.getDocuments(cbl_db, cbl_doc_ids)
+    assert len(cbl_docs) == 90, "The number of docs pull from SGW is incorrect"
+    if not cbl_action_before_init_replication:
+        for deleted_sg_doc in docs_to_delete_1:
+            assert deleted_sg_doc["id"] not in cbl_doc_ids, "deleted doc on SGW should not be pulled to CBL"
+
+    # 7. delete 15 more docs on SGW
+    docs_to_delete_2 = sg_added_docs[10:25]
+    sg_client.delete_docs(url=sg_url, db=sg_db, docs=docs_to_delete_2, auth=auth_session)
+    sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db, include_docs=True)["rows"]
+    assert len(sg_docs) == 75, "Expected number of docs is incorrect after doc deletion in sync-gateway"
+
+    # 8. verify SGW stats, cbl_replication_pull.rev_send_count sent to CBL is still (90 + 15 deleted) 105
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    replicator.stop(repl)
+
+    expvars = sg_client.get_expvars(sg_admin_url)
+    rev_send_count2 = expvars["syncgateway"]["per_db"][sg_db]["cbl_replication_pull"]["rev_send_count"]
+    assert rev_send_count2 == 115, "rev_send_count {} is incorrect".format(rev_send_count)
```

Comment:
isn't it 105 based on your summary steps? 90 + 15  should be 105. It is pull replicator only?

Reply:
corrected the number described in summary. 115 should be the correct value. for valid empty db, 1st time count=90, second time count=10(deleted before first replication)+15(deleted before second replication).